engine/src/main/java/org/terasology/engine/rendering/opengl/ScreenGrabber.java

```diff
@@ -101,13 +101,15 @@ public void saveScreenshot() {
         int width = sceneFinalFbo.width();
         int height = sceneFinalFbo.height();
 
+        Runnable task;
         if (savingGamePreview) {
-            GameScheduler.parallel().schedule(() -> saveGamePreviewTask(buffer, width, height));
+            task = () -> saveGamePreviewTask(buffer, width, height);
             this.savingGamePreview = false;
         } else {
-            GameScheduler.parallel().schedule(() -> saveScreenshotTask(buffer, width, height));
+            task = () -> saveScreenshotTask(buffer, width, height);
         }
 
+        GameScheduler.scheduleParallel("Write screenshot", task);
```

Comment:
would probably still opt to just use two calls to `scheduleParallel`. no need to be clever.